lib/action_dispatch/session/active_record_store.rb

```diff
@@ -67,7 +67,7 @@ def get_session(request, sid)
             # If the sid was nil or if there is no pre-existing session under the sid,
             # force the generation of a new sid and associate a new session associated with the new sid
             sid = generate_sid
-            session = @@session_class.new(:session_id => sid.private_id, :data => {})
+            session = session_class.new(:session_id => sid.private_id, :data => {})
```

Comment:
`session_class` is a `cattr_accessor`, meaning subclasses will fight over that variable. You want `class_attribute`.

Reply:
cattr_accessor actually generates [https://apidock.com/rails/Class/cattr_accessor](https://apidock.com/rails/Class/cattr_accessor) but agreed, I will change it to `self.class.session_class`

Reply:
updated it

Reply:
> cattr_accessor actually generates

That's not the problem, the problem is that it is backed by a class variable (`@@`) https://github.com/rails/rails/blob/522c86f35ccc80453ed9fb6ca8b394db321f9a69/activesupport/lib/active_support/core_ext/module/attribute_accessors.rb#L124

Reply:
> updated it

I didn't mean to chance for `self.class....` calls, but to declare the variable with `class_attribute`.

Reply:
thanks, never ran across `class_attribute` before